code/edge/src/main/java/com/adobe/marketing/mobile/NetworkResponseHandler.java

```diff
@@ -558,6 +558,15 @@ private void dispatchEventErrors(final JSONArray errorsArray, final boolean isEr
 
 			logErrorMessage(currentError, isError, requestId);
 
+			if (report != null) {
+				// Do not include eventIndex in the response
+				report.remove(EdgeJson.Response.EventHandle.EVENT_INDEX);
```

Comment:
should we try to remove it from the eventDataResponse directly? looks like the optTypedMap is returning the reference and not the new map for the selected key, but I am worried that if this behavior changes in the future it may break this.